betree/src/database/mod.rs

```diff
@@ -69,72 +63,21 @@ const DEFAULT_SYNC_INTERVAL_MS: u64 = 1000;
 
 type Checksum = XxHash;
 
-type ObjectPointer = data_management::impls::ObjectPointer<Checksum, DatasetId, Generation>;
-pub(crate) type ObjectRef = data_management::impls::ObjectRef<ObjectPointer>;
-type Object = Node<ObjectRef>;
+type ObjectPointer = data_management::ObjectPointer<Checksum>;
+pub(crate) type ObjectRef = data_management::impls::ObjRef<ObjectPointer>;
+pub(crate) type Object = Node<ObjectRef>;
+type DbHandler = Handler<ObjectRef>;
 
-pub(crate) type RootDmu = Dmu<
-    ClockCache<data_management::impls::ObjectKey<Generation>, RwLock<Object>>,
-    StoragePoolUnit<XxHash>,
-    Handler,
-    DatasetId,
-    Generation,
-    DmlMsg,
->;
+pub(crate) type RootSpu = StoragePoolUnit<XxHash>;
+pub(crate) type RootDmu =
+    Dmu<ClockCache<data_management::impls::ObjectKey<Generation>, RwLock<Object>>, RootSpu>;
 
 pub(crate) type MessageTree<Dmu, Message> =
     Tree<Arc<Dmu>, Message, Arc<TreeInner<ObjectRef, DatasetId, Message>>>;
 
 pub(crate) type RootTree<Dmu> = MessageTree<Dmu, DefaultMessageAction>;
 pub(crate) type DatasetTree<Dmu> = RootTree<Dmu>;
 
-/// This trait describes a multi-step database initialisation procedure.
```

Comment:
I can see the methods of DatabaseBuilder are moved to DatabaseConfiguration, please correct me if I am wrong. First, thank you for simplifying different structs (e.g. Database) by excluding the DatabaseBuilder as the template parameter from their declarations/definitions, as I think was a misuse of templates. However, one small remark, I think from the naming (or responsibility) perspective the current approach was fine, as "Builder" would initialize database objects and "Configuration" was just a data-structure holding the parameters from the config file. Converting the Builder to a static entity and passing DatabaseConfiguration object as a param could also have been an option here. Anyhow it is just a remark, thank you for the change :).